diff --git a/node_modules/shamirs-secret-sharing/random.js b/node_modules/shamirs-secret-sharing/random.js
index 398bf3b..a0f1535 100644
--- a/node_modules/shamirs-secret-sharing/random.js
+++ b/node_modules/shamirs-secret-sharing/random.js
@@ -1,4 +1,4 @@
-const { randomBytes } = require('crypto')
+const { randomBytes } = require('react-native-randombytes')
 
 function random(size) {
   const r = randomBytes(32 + size)
